### PR TITLE
MCOL-3915 Update columnstore install and uninstall procedure

### PR DESCRIPTION
**Pull request by jmrojas2332**
Columnstore installation now leaves mariadb in the same state (active or inactive) as it was before installation. Columnstore properly cleans up log/tmp/share files during columnstore-pre-uninstall w/o the warnings from rpm.

Also did some cleanup.

### CODE REVIEW

File: cmake/cpackEngineRPM.cmake
Comment (jmrojas2332) [inline cmake/cpackEngineRPM.cmake:89]: Typo here, partly explains the case David Hall had where old columnstore was not cleaned up properly.

File: oam/install_scripts/columnstore-post-install.in
Comment (jmrojas2332) [inline oam/install_scripts/columnstore-post-install.in:31]: startup mariadb if not already running. Need mariadb for some installation.

File: oam/install_scripts/columnstore-post-install.in
Comment (jmrojas2332) [inline oam/install_scripts/columnstore-post-install.in:78]: The formatting finally got to me so I did some cleanup since I was here

File: oam/install_scripts/columnstore-post-install.in
Comment (jmrojas2332) [inline oam/install_scripts/columnstore-post-install.in:199]: Let us not forget to stop mariadb if we started it.

File: oam/install_scripts/columnstore-pre-uninstall.in
Comment (jmrojas2332) [inline oam/install_scripts/columnstore-pre-uninstall.in:24]: clearShm would have already been called during systemctl stop so only need this in here w/o systemd

File: oam/install_scripts/columnstore-pre-uninstall.in
Comment (jmrojas2332) [inline oam/install_scripts/columnstore-pre-uninstall.in:31]: Let us make sure that an old systemd gets rid of running ProcMon/ProcMgr.
columnstore-pre-uninstall is also called during installation as a "pre-install" script.

File: oam/install_scripts/columnstore-pre-uninstall.in
Comment (jmrojas2332) [inline oam/install_scripts/columnstore-pre-uninstall.in:47]: No need for this. Will already be cleaned up during yum remove mariadb-columnstore

File: oam/install_scripts/columnstore-pre-uninstall.in
Comment (jmrojas2332) [inline oam/install_scripts/columnstore-pre-uninstall.in:110]: only file left in /usr/share/columnstore after package removal. I don't really think we need this if we do a removal.

File: oam/install_scripts/mcs-stop-controllernode.sh.in
Comment (jmrojas2332) [inline oam/install_scripts/mcs-stop-controllernode.sh.in:7]: Be safe here. If pgrep -x controllernode returns nothing it will not treat it as an empty string

File: oam/install_scripts/post-mysql-install
Comment (jmrojas2332) [inline oam/install_scripts/post-mysql-install:79]: Moved a variation of this to columnstore-post-install w/o skip-grant-tables

File: oamapps/postConfigure/helpers.cpp
Comment (jmrojas2332) [inline oamapps/postConfigure/helpers.cpp:258]: minor cleanup

File: oamapps/postConfigure/postConfigure.cpp
Comment (jmrojas2332) [inline oamapps/postConfigure/postConfigure.cpp:141]: cleanup postconfigure some more

File: oam/install_scripts/columnstore-post-install.in
Comment (jmrojas2332) [reply oam/install_scripts/columnstore-post-install.in:199]: Need to test this on non-systemd mariadb. Don't merge unless you're sure this works. we are starting mysqld_safe so i need to make sure this doesn't need to change. Will test when back at work.

File: dbcon/mysql/install_mcs_mysql.sh.in
Comment (drrtuy) [inline dbcon/mysql/install_mcs_mysql.sh.in]: This doesn't seem right. You need to replace /tmp/123.sql with the file or sqls we are executing.
I wonder if this works?

File: oam/install_scripts/columnstore-post-install.in
Comment (drrtuy) [inline oam/install_scripts/columnstore-post-install.in:122]: JFYI This script now effectively does nothing. We will put our automation start/stop in one and use it here.

File: oam/install_scripts/columnstore-post-install.in
Comment (jmrojas2332) [reply oam/install_scripts/columnstore-post-install.in:199]: Confirmed system works as expected when mariadb system unit does not exist

**Comment by jmrojas2332** [comment]
I have now tested on both a system with a mariadb systemd unit and without a mariadb systemd unit.

**Review by drrtuy — CHANGES_REQUESTED** [review]

File: oam/install_scripts/columnstore-post-install.in
Comment (drrtuy) [inline oam/install_scripts/columnstore-post-install.in:24]: I found out that service might be masked(/etc/systemd/system/mariadb.service is a symbolic link to /dev/null) but `systemctl cat mariadb.service` returns 0. I'll fix this after we merge this one.

File: dbcon/mysql/install_mcs_mysql.sh.in
Comment (jmrojas2332) [reply dbcon/mysql/install_mcs_mysql.sh.in]: I have applied the proper fix in the most recent commit and have tested install_mcs_mysql is working by removing schema associated with columnstore and reinstalling to confirm columnstore-related schema is created

**Review by drrtuy — APPROVED** [review]

**Pull request merged**
